pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -558,6 +585,14 @@ stdenv.mkDerivation ({
     runHook postInstall
   '';
 
+  ${if doInstallIntermediates then "installIntermediatesPhase" else null} = ''
+    runHook preInstallIntermediates
+    installIntermediatesDir=${if enableSeparateIntermediatesOutput then "$intermediates" else "$out"}
+    mkdir -p $installIntermediatesDir/dist
+    cp -r dist/build $installIntermediatesDir/dist
```

Comment:
I wonder if we can find a better dir for this (under share?) that includes pname and version so there are no collisions if `enableSeparateIntermediatesOutput` is false and it is installed by accident.

Reply:
Good idea. I updated this to install intermediates to `share/haskell/${ghc.version}/${pname}-${version}/dist`. I think it might be better to key them under `${pname}` rather than `${pname}-${version}`, though — we wouldn't want to have an incremental build cache miss just because a version number incremented. Let me know what you think.